test/e2e_node/pod_host_ips.go

```diff
@@ -76,11 +84,16 @@ var _ = common.SIGDescribe("DualStack Host IP [NodeFeature:PodHostIPs] [Feature:
 			podClient := e2epod.NewPodClient(f)
 			p := podClient.CreateSync(ctx, pod)
 
+			_ = e2epod.WaitForPodCondition(ctx, f.ClientSet, p.Namespace, p.Name, "PodHostIPs", 30*time.Second, func(pod *v1.Pod) (bool, error) {
+				return pod.Status.HostIPs == nil, nil
+			})
+			p, err := podClient.Get(ctx, pod.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err)
 			gomega.Expect(p.Status.HostIP).ShouldNot(gomega.BeEquivalentTo(""))
 			gomega.Expect(p.Status.HostIPs).Should(gomega.BeNil())
```

Comment:
I don't think you need this async check, the CreateSync method already waits for the pod to be ready and at taht point it should have the ips set.

Just remove the async checks added, only add the tag for the security thing

Reply:
great, thanks for pointing out

Reply:
in that case, you need to move the assertion in the async loop, don't need to assert twice

```
			err := e2epod.WaitForPodCondition(ctx, f.ClientSet, p.Namespace, p.Name, "PodHostIPs", 30*time.Second, func(pod *v1.Pod) (bool, error) {
                                 if pod.Status.HostIP == "" {
                                         return false, nil
                                 }
				return len(pod.Status.HostIPs) >0 , nil
			})
-			p, err := podClient.Get(ctx, pod.Name, metav1.GetOptions{})
			framework.ExpectNoError(err)
-			gomega.Expect(p.Status.HostIP).ShouldNot(gomega.BeEquivalentTo(""))
-			gomega.Expect(p.Status.HostIPs).Should(gomega.BeNil())
```

Reply:
just add
```
			err := e2epod.WaitForPodCondition(ctx, f.ClientSet, p.Namespace, p.Name, "PodHostIPs", 30*time.Second, func(pod *v1.Pod) (bool, error) {
                                 if pod.Status.HostIP == "" {
                                         return false, nil
                                 }
				return len(pod.Status.HostIPs) >0 , nil
			})
			if err != nil {
                                framework.Failf("timeout waiting for Pod %s to get HostIPs", pod.Name, err)
                        }

```

and remove 
```
			p, err := podClient.Get(ctx, pod.Name, metav1.GetOptions{})
			framework.ExpectNoError(err)
			gomega.Expect(p.Status.HostIP).ShouldNot(gomega.BeEquivalentTo(""))
			gomega.Expect(p.Status.HostIPs).Should(gomega.BeNil())
- ```